packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -511,26 +517,44 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         .domain([0, this._yMax])
         .range([0, containerHeight - this.margins.bottom! - this.margins.top!]);
       return { xBarScale, yBarScale };
-    } else {
-      const xBarScale = d3ScaleBand()
-        .domain(this._xAxisLabels)
-        .range(
-          this._isRtl
-            ? [containerWidth - this.margins.right! - this._domainMargin, this.margins.left! + this._domainMargin]
-            : [this.margins.left! + this._domainMargin, containerWidth - this.margins.right! - this._domainMargin],
-        )
-        .paddingInner(2 / 3);
+    }
+    if (this._xAxisType === XAxisTypes.DateAxis) {
+      const sDate = d3Min(this._points, (point: IVerticalBarChartDataPoint) => {
+        return point.x as Date;
+      })!;
+      const lDate = d3Max(this._points, (point: IVerticalBarChartDataPoint) => {
+        return point.x as Date;
+      })!;
+      const xBarScale = d3ScaleTime()
+        .domain([sDate, lDate])
+        .range([
+          this.margins.left! + this._domainMargin + this._barWidth / 2,
+          containerWidth - this.margins.right! - this._barWidth / 2 - this._domainMargin,
+        ]);
 
       const yBarScale = d3ScaleLinear()
```

Comment:
>yBarScale [](http://example.com/codeflow?start=12&length=9)

yBarScale code is repeating thrice. We should have it only once.

Reply:
Done